### DISCUSSION TITLE
Why is `stroked` a float and `filled` a boolean in the ScatterplotLayer

### DISCUSSION THREAD

**Discussion by jeffpeck10x** [visgl/deck.gl#6756] [General]
I noticed in the ScatterplotLayer that `stroked` is passed as a float:
```
uniform float stroked;
uniform bool filled;
```
https://github.com/visgl/deck.gl/blob/8.7-release/modules/layers/src/scatterplot-layer/scatterplot-layer-vertex.glsl.js#L41

Is that purely for the convenience of being able to do this:
```
innerUnitRadius = 1.0 - stroked * lineWidthPixels / outerRadiusPixels;
```

without having to cast `float(stroked)`?

**Comment by Pessimistress** [comment]
Yes I believe so - there are some legacy code in the shaders that are not super consistent. Feel free to open a PR if it bothers you :)

**Comment by jeffpeck10x** [comment]
Doesn't bother me at all. Just looking at it for ideas in a custom layer and wanted to know if there was some special reason (like, "casting bool to float is expensive", or similar.)
